include/openssl/core.h

```diff
@@ -75,66 +75,39 @@ struct ossl_algorithm_st {
 struct ossl_param_st {
     const char *key;             /* the name of the parameter */
     unsigned int data_type;      /* declare what kind of content is in buffer */
-    void *buffer;                /* value being passed in or out */
-    size_t buffer_size;          /* buffer size */
-    size_t *return_size;         /* OPTIONAL: address to content size */
+    void *buffer;                /* value being passed */
+    size_t size;                 /* buffer size */
+    size_t used;                 /* amount used (if not implied by data_type) */
```

Comment:
This implies that the OSSL_PARAM itself is modifiable by called functions. The reason I would rather not is to try to protect against badly written code that might corrupt the OSSL_PARAM.

Reply:
If you use the API those mistakes won't happen.  If you dn't use the API, well, it's a public structure so you can't prevent them anyway.